server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -324,4 +350,14 @@ protected void doXContent(XContentBuilder builder, ToXContent.Params params) thr
             builder.field("numeric_detection", numericDetection.value());
         }
     }
+
+    /**
+     * Returns a view of all full field paths to mappers
+     *
+     * @param path The path.
+     * @return The {@link Mapper} or null if none was found.
+     */
+    public Mapper mapperForPath(final String path) {
+        return this.pathToMapper.get(path);
+    }
```

Comment:
only really used by the alias to (validate) and fetch the target field its aliased too etc.